com.unity.multiplayer.mlapi/Tests/Editor/NetworkBehaviourTests.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using NUnit.Framework;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace MLAPI.EditorTests
+{
+    public class NetworkBehaviourTests
+    {
+        [Test]
+        public void HasNetworkObjectTest()
+        {
+            var gameObject = new GameObject(nameof(NetworkObject));
+            var networkBehaviour = gameObject.AddComponent<EmptyNetworkBehaviour>();
+            
+            Assert.That(networkBehaviour.HasNetworkObject, Is.False);
+
+            var networkObject = gameObject.AddComponent<NetworkObject>();
+            
+            Assert.That(networkBehaviour.HasNetworkObject, Is.True);
+            
+            Object.DestroyImmediate(networkObject);
+            
+            Assert.That(networkBehaviour.HasNetworkObject, Is.False);
+        }
+        
+        [Test]
+        public void AccessNetworkObjectTest()
+        {
+            var gameObject = new GameObject(nameof(NetworkObject));
+            var networkBehaviour = gameObject.AddComponent<EmptyNetworkBehaviour>();
+
+            // TODO: Do we really want to throw here?
+            // Future API change: return null
+            Assert.Throws<NullReferenceException>(() =>
+            {
+                var x = networkBehaviour.NetworkObject;
+            });
```

Comment:
yeah, we should probably not throw but return `null` there instead but obviously not in this PR

Reply:
Please create a quick JIRA ticket so as to not rely on our tribal knowledge & likelihood we'll scan for TODOs someday